packages/toolpad-app/src/components/AppEditor/PageEditor/QueryEditor.tsx

```diff
@@ -283,11 +284,36 @@ function QueryNodeEditorDialog<Q, P>({
     { retry: false },
   );
 
+  const untransformedQueryPreview = client.useQuery(
+    'execQuery',
+    input
+      ? [
+          appId,
+          {
+            ...input,
+            attributes: {
+              ...input.attributes,
+              transform: { value: '', type: 'const' },
+              transformEnabled: { value: false, type: 'const' },
+            },
+          },
+          paramsObject,
+        ]
+      : null,
+    { retry: false },
+  );
+
   const handleUpdatePreview = React.useCallback(() => {
     setPreviewQuery(input);
     setPreviewParams(paramsObject);
   }, [input, paramsObject]);
 
+  const handleTransformEditorFocus = React.useCallback(() => {
+    if (input) {
```

Comment:
Interesting, does it try to refetch even though the query is disabled (because we provided `null` as the key)?

Reply:
Yes, that's exactly what is happening